src/nspawn/nspawn.c

```diff
@@ -378,6 +379,34 @@ static void parse_share_ns_env(const char *name, unsigned long ns_flag) {
         arg_clone_ns_flags = (arg_clone_ns_flags & ~ns_flag) | (r > 0 ? 0 : ns_flag);
 }
 
+
+static void parse_mount_settings_env(void) {
+        int r;
+        const char *e;
+
+        e = getenv("SYSTEMD_NSPAWN_API_VFS_WRITABLE");
+        if (!e)
+                return;
+
+        if (streq(e, "network")) {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_NETNS;
+                return;
+        }
+
+        r = getenv_bool("SYSTEMD_NSPAWN_API_VFS_WRITABLE");
+        if (r < 0) {
+                log_warning_errno(r, "Failed to parse SYSTEMD_NSPAWN_API_VFS_WRITABLE from environment, ignoring.");
+                return;
+        } else if (r > 0) {
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_NETNS;
+        } else {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_NETNS;
```

Comment:
Is this intentational that lines 403 and 406 are the same? If yes, it should be moved below the else block.